openstack/networking/v2/subnets/requests.go

```diff
@@ -108,6 +108,7 @@ type CreateOpts struct {
 	TenantID        string
 	AllocationPools []AllocationPool
 	GatewayIP       string
+	NoGateway       bool
```

Comment:
Right. The issue I ran into is that an omission of `EnableDHCP` was turning DHCP on. I did a scan through the Neutron code and couldn't find where DHCP would be turned on if `enable_dhcp` was omitted, but it was.

By having the direct value of `EnableDHCP` passed through, that would have been avoided. See here: https://github.com/hashicorp/terraform/pull/6052/files

So I'm wondering if `NoGateway` should be `bool` or `*bool`. IMO, the former makes sense and is more intuitive, but having a `bool` and `*bool` attribute in the struct seems to complicate the structure.

Thoughts?